### ISSUE TITLE
Mocked mongoose#connect is not source-compatible with mongoose

### ISSUE THREAD

**Issue by alex94puchades** [mccormicka/Mockgoose#130]
The following code runs just fine with mongoose, but throws a ConnectionError with mockgoose:

```js
var util = require('util');
var mongoose = require('mongoose');
var mockgoose = require('mockgoose');
mockgoose(mongoose);

mongoose.connect('mongodb://localhost/test');
var Test = mongoose.model('Test', new mongoose.Schema({
	foo: String,
}));
	
Test.find(function(err, results) { // [1]
	if (err) return console.log(err.message);
	console.log(util.inspect(results));
	mongoose.disconnect();
});
```

It seems ```handleConnection``` is called too late, and ```conn._mockReadyState``` is not set when [1] is called. The following code works, but in practice it means mockgoose is not source-code compatible with mongoose, which is a major issue.

```js
var util = require('util');
var mongoose = require('mongoose');
var mockgoose = require('mockgoose');
mockgoose(mongoose);

mongoose.connect('mongodb://localhost/test', function(err) {
	if (err) return console.log(err.message);
	
	var Test = mongoose.model('Test', new mongoose.Schema({
		foo: String,
	}));
	
	Test.find(function(err, results) {
		if (err) return console.log(err.message);
		console.log(util.inspect(results));
		mongoose.disconnect();
	});
});
```
In addition, the script hangs at the end, it seems ```mongoose.disconnect``` is not working either.

**Comment by zxqfox** [comment]
This code just stuck with mongoose 4.1.1 ;-\

**Comment by winfinit** [comment]
this should be fixed in 5.0
